### Model Validation Technique
There are 3 Technique

1.Train & Test Split

2.K-fold Cross Validation

3.Leave One Out Cross Validation

#### Using a Logistic Model 

#### 1.Evaluate using a train and a test set

In [1]:
# Evaluate using a train and a test set

#Importing the Required Libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn import preprocessing

import warnings
warnings.filterwarnings("ignore")  #--to ignore warnings

#Load the data set
cancer = pd.read_csv("C:/Users/Akaash/Downloads/cancer.csv")

#Complete Cancer dataset - applying lable encoder to species column - (Y Variable is Categorial)
label_encoder = preprocessing.LabelEncoder()
cancer['diagnosis'] = label_encoder.fit_transform(cancer['diagnosis']) 

# Dividing our data into input and output variables 
X = cancer.iloc[:,1:]
Y = cancer.iloc[:,0]

#Splitting the Dataset into Train & Test 
test_size = 0.33
seed = 7
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=test_size,random_state=seed)

#Training the Model
model = LogisticRegression()
model.fit(X_train, Y_train)

#Predicting & Getting the Model Score
result = model.score(X_test, Y_test)

#Calculating the Accuracy of the Model
result*100.0

95.2127659574468

Inference: With Train & Test Split The Accuarcy has come 95.21%

#### 2.Evaluate using K-Fold Cross Validation

In [2]:
# Evaluate using Cross Validation

#Importing the Required Libraries
import pandas as pd
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn import preprocessing

#Load the data set
cancer = pd.read_csv("C:/Users/Akaash/Downloads/cancer.csv")

#Complete Cancer dataset - applying lable encoder to species column - (Y Variable is Categorial)
label_encoder = preprocessing.LabelEncoder()
cancer['diagnosis'] = label_encoder.fit_transform(cancer['diagnosis']) 

# Dividing our data into input and output variables 
X = cancer.iloc[:,1:]
Y = cancer.iloc[:,0]

# Creating K-Fold Instance
num_folds = 10
seed = 7
kfold = KFold(n_splits=num_folds, random_state=seed,shuffle=True)

#Training the Model 
model = LogisticRegression(max_iter=400)

#Predicting & Getting the Model Score with kfold
results = cross_val_score(model, X, Y, cv=kfold)


In [3]:
# Accuracy for Each K-Fold
results

array([0.9122807 , 0.98245614, 0.96491228, 0.94736842, 0.94736842,
       0.89473684, 0.96491228, 0.98245614, 0.92982456, 0.94642857])

In [4]:
#Average of All k-fold Accuracy to get Final model Accuary
results.mean()*100.0

94.72744360902254

Inference: The K-fold gave the final Accuracy of 94.72%

#### 3.Evaluate using Leave One Out Cross Validation

In [5]:
# Evaluate using Leave One Out Cross Validation

#Importing the Required Libraries
import pandas as pd
from sklearn.model_selection import LeaveOneOut
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn import preprocessing

#Load the data set
cancer = pd.read_csv("C:/Users/Akaash/Downloads/cancer.csv")

#Complete Cancer dataset - applying lable encoder to species column - (Y Variable is Categorial)
label_encoder = preprocessing.LabelEncoder()
cancer['diagnosis'] = label_encoder.fit_transform(cancer['diagnosis']) 

# Dividing our data into input and output variables 
X = cancer.iloc[:,1:]
Y = cancer.iloc[:,0]

# Creating K-Fold Instance
loocv = LeaveOneOut()

#Training the Model
model = LogisticRegression(max_iter=300)

#Predicting & Getting the Model Score with Leave One Out Cross Validation
results = cross_val_score(model, X, Y, cv=loocv)

In [6]:
#Average of All LooCV Accuracy to get Final model Accuary
results.mean()*100.0

94.90333919156416

Inference: With Leave One Out Cross Validation The Accuarcy has come 94.90%